### ISSUE TITLE
FixedMassABM performance issues

### ISSUE THREAD

**Issue by mdotend** [JuliaDynamics/Agents.jl#761]
**Describe the bug**
I implemented the Schellingers model (though with 3 groups) with the FixedMassABM as it was recommended in the documentation, but I fail to see performance benefits. In fact, the UnkillableABM implementation seems to be faster by at least 5 times and has less allocations.

I am not sure, whether this is a problem of my initialization or the benchmarking, but the difference is odd:

Result of the unkillable Model:   240.832 ms (1 allocation: 96 bytes)
Result of the fixed size Model:   2.788 s (30461001 allocations: 750.90 MiB)

**Minimal Working Example**
```julia
using Agents, Random
using BenchmarkTools

@agent SchellingAgent GridAgent{2} begin
	mood::Bool
	group::Int
end

function initUnkillableModel(; total_agents = 75, griddims = (10, 10), min_to_be_happy = 3, seed = 12345)
    space = GridSpaceSingle(griddims, periodic = false)
    properties = Dict(:min_to_be_happy => min_to_be_happy)
    rng = Random.Xoshiro(seed)

    model = UnkillableABM(SchellingAgent, space; properties, rng, scheduler = Schedulers.Randomly())
    for n in 1:total_agents
        agent = SchellingAgent(n, (1, 1), false, n % 3)
        add_agent_single!(agent, model)
    end

    return model
end

function initFixedModel(; total_agents = 75, griddims = (10, 10), min_to_be_happy = 3, seed = 12345)
    space = GridSpaceSingle(griddims, periodic = false)
    properties = Dict(:min_to_be_happy => min_to_be_happy)
    rng = Random.Xoshiro(seed)

    agentVector = Vector{SchellingAgent}()
    for n in 1:total_agents
        agent = SchellingAgent(n, (1, 1), false, n % 3)
        push!(agentVector, agent)
    end
    model = FixedMassABM(agentVector, space; properties, rng, scheduler = Schedulers.Randomly()) 
return model
end

fixedModel = initFixedModel(total_agents = 7500, griddims = (100,100))
unkillableModel = initUnkillableModel(total_agents = 7500, griddims = (100,100))

function agent_step!(agent, model)
	minhappy = model.min_to_be_happy
	count_neighbors_same_group = 0
        nagents(model)*5
	for neighbour in nearby_agents(agent, model)
		if agent.group == neighbour.group
			count_neighbors_same_group += 1 
		end
	end
	if count_neighbors_same_group ≥ minhappy
        agent.mood = true
    else
        agent.mood = false
        move_agent_single!(agent, model)
    end
    return
end

print("Result of the unkillable Model: ")
@btime step!(unkillableModel, agent_step!, 500)

print("Result of the fixed size Model: ")
@btime step!(fixedModel, agent_step!, 500)
```

**Agents.jl version**
```
Status `~/.julia/environments/v1.8/Project.toml`
  [46ada45e] Agents v5.8.0
```

**Comment by mdotend** [comment]
So I did some digging and when I change the type of the vector from SizedVector to SVector in the model_concrete.jl file, I get at least the same execution time and more reasonable allocations.  
```
Result of the unkillable Model:   46.191 ms (1 allocation: 96 bytes)
Result of the fixed size Model:   46.490 ms (1 allocation: 13.44 KiB)
```
I do not understand the StaticArrays package well enough to spot the difference, but in their documentation they state, that the benefit is only for Vectors with a length less than 100, so the actual performance might be only measurable in small models.


PS: The change to the SVector type does not fix the issue with plotting though...

**Comment by Datseris** [comment]
Models with less than 100 agents are anyways so fast that we don't care about using static vectors. I'll make a PR that deprecates the FixedMassABM. It was my fault in the first place that we put it in: I suggested it as an option but we never actually measured whether it has performance benefits.
